### PR TITLE
Update kubernetes version used for travis

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

- Enhancement / new feature

### Description

This PR updates travis to use more recent versions of kubectl, minikube and specifically kubernetes 1.15. The fixes to the ITs are necessary because those tests exit early on current 3.9 travis build, but don't work on minikube (they do on oc cluster up 3.11). There reason is that the patch wasn't actually changing anything, so the update in those tests did not result in increment of metadata.resourceVersion, which the test relied on in order to get and test for a resource conflict. Making different changes to the json ensures the patch does indeed result in a new version on the server.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
+dist: xenial
```
Comment (ppatierno) [inline .travis.yml]: out of curiosity, what's the need for changing the distro?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java [hunk 2]
```diff
@@ -305,7 +304,7 @@ public void testUpdateStatusWhileResourceUpdated(TestContext context)    {
         log.info("Updating resource");
         KafkaConnect updated = new KafkaConnectBuilder(kafkaConnectOperator.get(namespace, RESOURCE_NAME))
                 .editSpec()
-                .withLogging(new InlineLogging())
+                .withNewLivenessProbe().withInitialDelaySeconds(14).endLivenessProbe()
```
Comment (ppatierno) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java]: is this related to this PR or you just noticed that? and why removing logging and adding probes?

Comment (tombentley) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java]: Well in testing this PR I discovered that the tests failed on minikube with kubernetes 1.14, but passed on `oc cluster up`. The tests didn't fail before on Travis because Travis was using 1.9, but these tests exit early when kubernetes <1.11 is detected.

I _think_ the reason the the logging change doesn't work is that, for whatever reason, kubernetes doesn't see the patch as causing an update. In certainly doesn't result in the resourceVersion getting incremented, which is what the test depends on (ultimately we're trying to get an resource version conflict).

File: .travis.yml (hunk 1 above)
Comment (tombentley) [reply .travis.yml]: See comment about xenial [here](https://github.com/LiliC/travis-minikube/blob/master/.travis.yml) (which is what I based these changes on).
